src/containers/ToDoList/ToDoList.jsx

```diff
@@ -103,54 +111,77 @@ export const ToDoList = () => {
     ? tasks.filter(taskFilters[filteringType])
     : tasks;
 
+  const handleChangeMode = () => {
+    setIsIconChanging((prevIcon) => !prevIcon);
+    setIsDarkTheme((prevTheme) => !prevTheme);
+    document.body.classList.toggle("light-theme", !isDarkTheme);
+  };
+
   return (
     <>
-      <StyledWrapperNewTask>
-        <button onClick={handleAddNewTask}>
-          <AddIcon />
-        </button>
-        <input
-          type="text"
-          placeholder="Create a new todo..."
-          value={newTaskInput}
-          onKeyDown={handleKeyDown}
-          onChange={(e) => setNewTaskInput(e.target.value)}
-        />
-      </StyledWrapperNewTask>
-
-      <StyledWrapper>
-        {filteredTasks.map((task) => (
-          <Task
-            id={task.id}
-            text={task.text}
-            checked={task.checked}
-            onCheck={handleTaskMark}
-            onDelete={handleDeleteTask}
+      <ThemeProvider theme={isDarkTheme ? lightTheme : darkTheme}>
+        <StyledWrapperTitle>
+          <h1>Todo</h1>
+          <button onClick={handleChangeMode}>
+            {isIconChanging ? (
+              <DarkModeIcon style={{ fontSize: 40, color: "#fff" }} />
+            ) : (
+              <LightModeIcon style={{ fontSize: 40, color: "#fff" }} />
+            )}
```

Comment:
Above - before the `return` you could do:

```
const ThemeChangingIcon = isDarkTheme ? LightModeIcon : DarkModeIcon;
```

and then this piece would become:

```suggestion
              <ThemeChangingIcon style={{ fontSize: 40, color: "#fff" }} />
```